lib/nexpose/scan.rb

```diff
@@ -407,9 +407,10 @@ def import_scan(site_id, zip_file)
       data = Rex::MIME::Message.new
       data.add_part(site_id.to_s, nil, nil, 'form-data; name="siteid"')
       data.add_part(session_id, nil, nil, 'form-data; name="nexposeCCSessionID"')
-      scan = ::File.new(zip_file, 'rb')
-      data.add_part(scan.read, 'application/zip', 'binary',
-                    "form-data; name=\"scan\"; filename=\"#{zip_file}\"")
+      ::File.new(zip_file, 'rb') do |scan|
```

Comment:
Just remembered `File.new` doesn't accept a block, we need to use `File.open` instead.

```
(irb):2: warning: File::new() does not take block; use File::open() instead
```

As for using `::File` over `File` I assume at one point (if not still the case) we had a conflicting constant in the Nexpose module namespace. Using `::File` is fine and leaving it alone doesn't affect this fix.